maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -481,6 +483,14 @@ class IOSDriver(
         throw TimeoutException("Maestro iOS driver did not start up in time")
     }
 
+    private fun <T> runDeviceCall(call: () -> T): T {
+        return try {
+            call()
+        } catch (appCrashException: IOSDeviceErrors.AppCrash) {
+            throw MaestroException.AppCrash(appCrashException.errorMessage)
+        }
```

Comment:
Right now for ease I have added one of the errors which can propagate from xctest layer but this should also be extended for the errors that can be thrown from SimctlIOSDevice.

Reply:
What happens in the event of another exception being thrown?

Reply:
If its not converted to a known exception then it will lead to cancellation.